### ISSUE TITLE
IPMI sol activate command is failing

### ISSUE THREAD

**Issue by rahulmah** [openbmc/phosphor-host-ipmid#174]
IPMI sol activate command is failing with message `No response activating SOL payload` after some time.

```
bash-4.2$ ipmitool -I lanplus -C 17 -N 3 -p 623 -U root -P 0penBmc -H <BMC_IP> sol activate usesolkeepalive
Error: No response activating SOL payload


bash-4.2$ ipmitool -I lanplus -C 17 -N 3 -p 623 -U root -P 0penBmc -H <BMC_IP> sol info
Info: SOL parameter 'Nonvolatile Bitrate (5)' not supported
Info: SOL parameter 'Volatile Bitrate (6)' not supported
Set in progress                 : set-complete
Enabled                         : true
Force Encryption                : true
Force Authentication            : true
Privilege Level                 : USER
Character Accumulate Level (ms) : 100
Character Send Threshold        : 1
Retry Count                     : 7
Retry Interval (ms)             : 100
Volatile Bit Rate (kbps)        : IPMI-Over-Serial-Setting
Non-Volatile Bit Rate (kbps)    : IPMI-Over-Serial-Setting
Payload Channel                 : 1 (0x01)
Payload Port                    : 623
bash-4.2$

```

BMC FW:
```
root@witherspoon:~# cat /etc/os-release
ID=openbmc-openpower
NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro)"
VERSION="2.10.0-rc1"
VERSION_ID=2.10.0-rc1-dirty
PRETTY_NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro) 2.10.0-rc1"
BUILD_ID="2.10.0-rc1"
OPENBMC_TARGET_MACHINE="witherspoon"
```

**Comment by JohnChung93** [comment]
Following commit cause this issue.
https://github.com/openbmc/phosphor-net-ipmid/commit/29086950c2300fe8bd5791896b1209a31a1c93e6

DBUS service missing that cause IPMI command response failed.
Following function could not get exist dbus service.

`            solService =
                ipmi::getService(dbus, solInterface, solPathWitheEthName);
`
You could refer following file for creating DBUS service which SOL needed.

https://github.com/openbmc/openbmc/blob/master/meta-quanta/meta-olympus-nuvoton/recipes-phosphor/settings/phosphor-settings-manager/sol-default.override.yml

It's works for me.
Hope this could fix your problem.

**Comment by leiyu-bytedance** [comment]
> Following commit cause this issue.
> [openbmc/phosphor-net-ipmid@2908695](https://github.com/openbmc/phosphor-net-ipmid/commit/29086950c2300fe8bd5791896b1209a31a1c93e6)
> 
> DBUS service missing that cause IPMI command response failed.
> Following function could not get exist dbus service.
> 
> ` solService = ipmi::getService(dbus, solInterface, solPathWitheEthName);`
> 
> You could refer following file for creating DBUS service which SOL needed.
> 
> https://github.com/openbmc/openbmc/blob/master/meta-quanta/meta-olympus-nuvoton/recipes-phosphor/settings/phosphor-settings-manager/sol-default.override.yml
> 
> It's works for me.
> Hope this could fix your problem.

With the fix, we still hit sol issue, see https://github.com/openbmc/phosphor-net-ipmid/issues/19 for details.
@JohnChung93 Did you meet the issue?

**Comment by JohnChung93** [comment]
> With the fix, we still hit sol issue, see [openbmc/phosphor-net-ipmid#19](https://github.com/openbmc/phosphor-net-ipmid/issues/19) for details.
> @JohnChung93 Did you meet the issue?

@leiyu-bytedance  No, I only meet the issue "No response activating SOL payload" for my system.

After setting all of sol dbus related property, I did not hit any IPMI sol issue.

**Comment by csiegpman** [comment]
I modified it according to leiyu-bytedance instructions, and then I can see the bus generated under bmc, but on the host, execute bash-4.2$ ipmitool -I lanplus -C 17 -N 3 -p 623 -U root -P 0penBmc -H <BMC_IP > sol activate usesolkeepalive
Still getting Error: No response activating SOL payload, how can I troubleshoot it?
![image](https://github.com/openbmc/phosphor-host-ipmid/assets/106588430/fb4cb114-752f-4d98-b6fc-5e64b1086694)
